app/controllers/lessons_controller.rb

```diff
@@ -105,13 +105,15 @@ def destroy
 
   # GET /l/future/slugs.json
   def future_lessons_slug
-    future_lesson_slugs = Lesson.future_lessons.pluck(:slug)
+    school = School.find_by(id: params[:school_id])
+    future_lesson_slugs = Lesson.future_lessons_for_school(school).pluck(:slug)
     render json: future_lesson_slugs
   end
 
   # GET /l/upcoming.json
   def upcoming
-    next_lesson = Lesson.future_lessons.first
+    school = School.find_by(id: params[:school_id])
+    next_lesson = Lesson.future_lessons_for_school(school).first
```

Comment:
There is already a `.future_lessons` and `.for_school` in the Lesson model, and I use them in `.future_lessons_for_school`. I moved checking for the presence of school to the model to keep the controller skinny and for easier testing.